python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,158 @@
+"""
+Table data structure that prints a nicely formatted table
+for an accumulator loop
+"""
+from __future__ import annotations
+
+import inspect
+import sys
+
+import astroid
+import tabulate
+
+
+def no_of_whitespaces(with_line: str) -> int:
+    """Return the number of whitespaces that are in the line of code
+    containing the with statement
+    """
+    blank_chars = 0
+    for char in with_line:
+        if char == " ":
+            blank_chars += 1
+        else:
+            break
+
+    return blank_chars
+
+
+def loop_string(full_string: list[str], no_whitespaces: int) -> str:
+    """Convert the list of lines that are strings from the beginning of the for loop
+    to the end of the for loop
+    """
+    endpoint = len(full_string)
+    for i in range(len(full_string)):
+        if full_string[i].strip() != "" and not full_string[i][no_whitespaces].isspace():
+            endpoint = i
+            break
+
+    return "\n".join(full_string[:endpoint])
+
+
+def setup_table(table: AccumulationTable) -> None:
+    """
+    Get the frame of the code containing the with statement, cut down the source code
+    such that it only contains the with statement and the accumulator loop and set up
+    the trace function to track the values of the accumulator variables during each iteration
+    """
+    class_frame = inspect.getouterframes(inspect.currentframe())[1].frame
```

Comment:
I'm not sure why this line is necessary. Seems like you should be able to get the `func_frame` directly from the call stack.